Source/Scenes/Onboarding/1_AskForAnalyticsPermissions/AskForAnalyticsPermissionsViewModel.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  AskForAnalyticsPermissionsViewModel.swift
+//  Zupreme
+//
+//  Created by Andrei Radulescu on 09/11/2018.
+//  Copyright © 2018 Open Zesame. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+import RxCocoa
+
+// MARK: - AnalyticsPermissionNavigation
+enum AskForAnalyticsPermissionsNavigation: String, TrackedUserAction {
+    case userOptedForAnalyticsPermissions
+}
+
+// MARK: - AnalyticsPermissionViewModel
+final class AskForAnalyticsPermissionsViewModel: AbstractViewModel<
+    AskForAnalyticsPermissionsNavigation,
+    AskForAnalyticsPermissionsViewModel.InputFromView,
+    AskForAnalyticsPermissionsViewModel.Output
+> {
+
+    private let useCase: OnboardingUseCase
+
+    init(useCase: OnboardingUseCase) {
+        self.useCase = useCase
+    }
+
+    override func transform(input: Input) -> Output {
+        
+        let readDisclaimerTrigger = input.fromView.readDisclaimerTrigger
+        
+        bag <~ [
```

Comment:
maybe we can change this to:

```swift
let userOptedForAnalyticsTrigger = Driver.merge(
	input.fromView.acceptTrigger.map { true },
	input.fromView.declineTrigger.map { false }
)

bag <~ [
	userOptedForAnalyticsTrigger.do(onNext: { [unowned self] in
           self.useCase.optedForAnalyticsPermissions(acceptsTracking: $0)
           self.stepper.step(.userOptedForAnalyticsPermissions)
     }).drive()
]
```

What do you think? Quite clean?

Reply:
Yes, it's much cleaner